### PR TITLE
feat: define an extensions section of the manifest

### PR DESCRIPTION
**Pull request by technosophos**
This defines a section of the manifest where arbitrary extensions can be added.

The goal of this section is to provide a specific location for extension data to be defined such that we don't end up like `package.json`, where tons of ad hoc extensions have been added to the root document.

### CODE REVIEW

**Comment by michelleN** [comment]
-1 for metadata, info
+1 for ext

**Comment by technosophos** [comment]
In the TUF spec, they use `custom`, which I sorta like as it gets right to the point.

**Comment by simongdavies** [comment]
+1 for custom

**Comment by garethr** [comment]
+1 for custom, although extensions also works for me

Note that this PR should also update the JSON Schema with the new fields.

File: 101-bundle-json.md [hunk 1]
```diff
@@ -453,5 +463,54 @@ The built-in actions (`install`, `upgrade`, `uninstall`) MUST NOT appear in the
...
+    }
+}
+```
+
+The format is:
+
+```json
+{
+    "extensions": {
+        "EXTENSION NAME": "ARBITRARY JSON DATA"
```
Comment (glyn) [inline 101-bundle-json.md]: This gives the impression that `"ARBITRARY JSON DATA"` must be double quote delimited. Is the intention that this should be a JSON-encoded string or can it be arbitrary "inline" JSON?

Comment (glyn) [reply 101-bundle-json.md]: I guess the examples resolve that question, but best to have some normative text too.

Comment (technosophos) [reply 101-bundle-json.md]: If only JSON had comments...

Comment (radu-matei) [reply 101-bundle-json.md]: No one say YAML. Please.

**Comment by technosophos** [comment]
Since the original PR was before the repo renames, I had to open a new version of this PR at #104. So that should supersede this one. I am closing this one.

**Pull request closed without merging**